lib/combine_pdf/page_methods.rb

```diff
@@ -63,15 +63,16 @@ def inject_page(obj, top = true)
       # injecting each of the values in the injected Page
       res = resources
       obj.resources.each do |key, new_val|
-        unless PDF::PRIVATE_HASH_KEYS.include? key # keep CombinePDF structual data intact.
-          if res[key].nil?
-            res[key] = new_val
-          elsif res[key].is_a?(Hash) && new_val.is_a?(Hash)
-            new_val = new_val[:referenced_object] || new_val
-            new_val.update (res[key][:referenced_object] || res[key]) # make sure the old values are respected
-            (res[key][:referenced_object] || res[key]).update new_val # transfer old and new values to the injected page
-          end #Do nothing if array - ot is the PROC array, which is an issue
-        end
+        # keep CombinePDF structural data intact.
+        next if PDF::PRIVATE_HASH_KEYS.include?(key)
```

Comment:
I'm just starting my first reading of the code, I need to read it again after reviewing what came before, so this comment might be mistaken. It's also here for me to remember.

I think we're loosing data here... one of the private keys is `:referenced_object` which exists (usually) when an object is actually a PDF reference (similar to a "pointer" in C) to a different object. Under the old logic, these would be reviewed whereas under the new logic they seem to be  ignored.